### ISSUE TITLE
figure out how to publish artifacts

### ISSUE THREAD

**Issue by wfurt** [wfurt/freebsd-bootstrap-cli#4]
I finally got Cirrus CI working. (removed everything and started over)
For runtime repo, we generally need everything from artifacts/packages/Release/Shipping/, most importantly dotnet-runtime-5.0.0-dev-freebsd-x64.tar.gz

cc: @am11

**Comment by am11** [comment]
Do we need these (intermediate?) raw bits from runtime in GH release artifacts or should we employee a FreeBSD task (which depends on Linux task), fetch the intermediate build output of cross compile from Linux task and prepare a real FreeeBSD source tarball there? Finally we can publish that source-build tarball to release artifacts (along with the working FreeBSD-flavored `dotnet(1)` and SDK).

**Comment by wfurt** [comment]
For master, the dotnet-runtime*.tgz is complete runtime e.g. enough for somebody to run app. 
And then we will need at least some of the *.nupkg. runtime.freebsd-x64* as minimum. I registered myget feed and it should be sufficient to post them right there. 

Then we should construct sdk. We can either plop Linux one and disable R2R or we can try to rebuild it. (that can be sorted later and I'm leaning toward first option + docs)

I did not think about generating  source-build tarball but that may work. That is actually bunch of binary as well, right? Would that be accepted as source for FreeBSD port package? Getting functional port would be my long term goal.

**Comment by am11** [comment]
Still thinking what value is CirrusCI bringing in all this? Given the distinguishing feature of CirrusCI vs. AzDO, Travis and AppVeyor etc. is the FreeBSD CI machine. For instance, we can use it for testing the binaries, the thing which we are currently missing from AzDO; before publishing anything.

**Comment by wfurt** [comment]
For me, the value would be availably of pre-build dotnet binaries and packages. I know various people (including me) did builds and testing but it may be nice to have something automated. Even if that is just interim.

**Comment by am11** [comment]
Makes sense. Given that, I think we can attempt incorporating the following bits in the existing .cirrus.yml as part of this issue:

* build the interested test suites in addition to binaries (if they are not already built)
* push `artifacts/packages/Release/Shipping/*.tar.gz` and test artifacts (location?) to local HTTP cache server (provided by CirrusCI)
* add FreeBSD  task, which pulls the artifacts from local cache server, extract them and runs the tests
* finally, publish artifacts (only `artifacts/packages/Release/Shipping/*.tar.gz` stuff) to GitHub release on tag push. we can also publish stuff to blob storage, myget, nuget etc.

after this, we can also try to do the following also in the CI, which you previously mentioned:

> We can either plop Linux one and disable R2R 

(which is something i have not done it myself so far and very keen to see in action 😃)

**Comment by wfurt** [comment]
I can do the sdk. (or verify that disabling R2R is sufficient). I you can work out the FreeBSD task, that would be great - even if it does not run the test quite yet. I think I know how to build the tests - I'll biggyback on Helix path which will construct runtime & zip files. 
Test runs produce XML result files. Normally we parse them and put them to Kusto database. Do you have idea how to publish test results? It would be great if somebody can look at a page or a file and see if everything passed and if not, what test failed. I think the processing can be just grep for now, but I'm not sure how to present it.
